services/bundle_analysis.py

```diff
@@ -368,22 +372,65 @@ def __init__(
         self.before = before
         self.branch = branch
 
+    def _compute_measurements(
+        self, measurable_name: str, measurable_ids: List[str]
+    ) -> dict[int, Iterable[dict[Any, Any]]]:
+        all_measurements = measurements_by_ids(
+            repository=self.repository,
+            measurable_name=measurable_name,
+            measurable_ids=measurable_ids,
+            interval=self.interval,
+            after=self.after,
+            before=self.before,
+            branch=self.branch,
+        )
+
+        print("what on earth", all_measurements)
+
+        # Carry over previous available value for start date if its value is null
+        for measurable_id, measurements in all_measurements.items():
+            if measurements[0]["timestamp_bin"] > self.after:
+                carryover_measurement = measurements_last_uploaded_by_before_date(
+                    repo_id=self.repository.repoid,
+                    measurable_name=measurable_name,
+                    measurable_ids=[measurable_id],
+                    start_date=self.after,
+                    branch=self.branch,
+                )
+
+                # There isn't any measurements before the start date range, it will just be null
+                if not carryover_measurement:
+                    continue
```

Comment:
Nit: change this to "if measurements: ..." and get rid of the continue/1 line of code

Reply:
No need to save a carryover with null values because there's a helper function that populations non-existent measurements with null values within the query date range.